firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionDataStoreConfigs.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.sessions
+
+import android.util.Base64
+import com.google.firebase.sessions.settings.SessionsSettings
+
+/**
+ * Util object for handling DataStore<Preferences> configs in multi-process apps safely.
+ *
+ * This can be removed when datastore-preferences:1.1.0 becomes stable.
+ */
+object SessionDataStoreConfigs {
+  /** Sanitized process name to use in config filenames. */
+  private val PROCESS_NAME =
+    Base64.encodeToString(
+      ProcessDetailsProvider.getProcessName().encodeToByteArray(),
+      Base64.NO_WRAP or Base64.URL_SAFE, // URL safe is also filename safe.
```

Comment:
Isn't a '/' url safe, but not filename safe? I don't think you can put that in a process name, but I wonder if we should wrap this reading/writing with a dynamic filename in a try catch block to make sure it doesn't cause the app to crash if it's poorly formed, or too long.

Reply:
I checked the doc on `URL_SAFE` it specifically says "URL and filename safe ... where - and _ are used in place of + and /" so it is safe.